Merge latest master

fingolfin opened the pull request:
@Pascalonyy are you still interested in this Origami work?

If so there are more changes / tweaks I could provide but that would require using the latest Oscar as basis.

fingolfin commented:
Ping @Pascalony (of course by now this PR itself is a bit stale but it should be no problem, another merge can be done at any time)

Pascalony commented:
@fingolfin we will update it soon, our last meetings were focused on the GAP package, but the work on this package will continue soon, too

HereAround committed:
Replace something.f by lift(something)

StevellM committed:
feat [`QuadFormAndIsom`]: more flexible inputs for functions with parameters (#5005)

jankoboehm committed:
Generic modules over the integers and fields (#4915)

fingolfin committed:
Replace unusual `getindex` methods (#5003)

... for IdealGens, ModuleGens, BiPolyArray

fingolfin committed:
Remove unused singular_assure; store ordering used for GB (#5006)

jamesnohilly committed:
Add natural_gset to action documentation (#5011)

mjrodgers committed:
Replace `AbstractAlgebra.combinations` (#4968)

fingolfin committed:
singular_assure: if I.isGB is set then I.ord must also be set (#5010)

fingolfin committed:
Remove singular_assure (#5012)

fingolfin committed:
Replace some uses of '@attr Any' (#5014)

HechtiDerLachs committed:
Repair simplification for free resolutions (#4982)

* Repair simplification for free resolutions.

* Repair the length for FreeResolution.

* Adapt canonical_bundle.

* Order the maps in the correct way.

* Use hard cut-off from what's already computed.

* Use new .length field in FreeResolution for Betti tables.

* Add kwarg for initial length to be computed with Schreyer's resolution.

* Revert "Repair the length for FreeResolution."

This reverts commit 345c56256b853dce093070e4f5605b88f1a821fe.

* Switch to old use of length in Betti tables.

* Remove field access for .length.

* Correct length method for FreeResolution.

* Make minimal_betti_table accept non-complete resolutions.

* Fix up docstring.

* Repair minimal_betti_table for complete resolutions.

* Unlazify minimized free resolutions.

* Export minimize.

* Fix doctests.

* Add intermediate patch to fix iterated resolutions.

* Revert "Add intermediate patch to fix iterated resolutions."

This reverts commit de5732978fd3e38b05bf4cfafedb7d89557d806c.

HereAround committed:
FTheoryTools: Improve documentation (#5009)

HereAround committed:
[FTheoryTools] Remember that ambient space for all QSMs is complete and smooth

HereAround committed:
Chern classes can be serialized

mjrodgers committed:
New implementation of `map_from_func` with changed argument order (#4988)

HechtiDerLachs committed:
Resolution cleanup, round two (#5018)

* Carry over printing through minimization.

* Add intermediate patch to fix iterated resolutions.

* Fix patch.

* Remove old code.

* Fix doctests.

* Add workaround for fres.

* Set complete flag.

* Add support for algorithm keyword in SimpleFreeResolution.

* Correct the passing of arguments.

* Remove debug comment.

* Adjust printing.

* Adjust tests to non-bound on resolutions.

* Adjust tests.

* Fix up printing of minimal_betti_table for non-complete resolutions.

* Hook up sres, too.

* Assure julia 1.6 compatibility.

* Assure compatibility with 1.6 again.

fingolfin committed:
Improve GModule support for ModuleFP (#4858)

HechtiDerLachs committed:
Improvements for sparse modules (#4955)

wdecker committed:
Adjust docu following PRs #4982, #5018 (#5025)

Adjust docu following PRs #4982, #5018

jankoboehm committed:
fix mres issue (#5027)

wdecker committed:
Update test following PR #5027 (#5029)

fingolfin committed:
Deprecate `vector_space_dimension`, use `vector_space_dim` instead (#5030)

lgoettgens committed:
Filter some more doctests for julia 1.13 (#5033)

And make two FTheoryTools doctests dict iteration order independent

lgoettgens committed:
BookTests: Use julia-version-specific testfile if available

lgoettgens committed:
Duplicate some booktest files with hash-related changes

lgoettgens committed:
Add missing space

fingolfin committed:
Rename MatrixGroup arg from x to G (#5038)

wdecker committed:
Free resolutions: correct length function (#5041)

* Free resolutions: correct length function

* correction

* Update src/Modules/UngradedModules/FreeResolutions.jl

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

---------

Co-authored-by: Max Horn <max@quendi.de>
Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

lgoettgens committed:
Fix more booktests with nightly and adjust path regex (#5044)

* Fix galoismod booktest with nightly

* booktest: adjust regex to filter methodinfo

---------

Co-authored-by: Benjamin Lorenz <lorenz@math.tu-berlin.de>

HereAround committed:
More features for FTheoryTools (#5031)

* Compute integral and rational coefficients for all G_4-fluxes.
* Upon construction, automatically express the defining cohomology class of each G_4-flux using the internally chosen generating set.
* Deprecate basis_of_h22_hypersurface in favor of gens_of_h22_hypersurface.
* Deprecate basis_of_h22_hypersurface_indices in favor of gens_of_h22_hypersurface_indices.

fingolfin committed:
Move serialization code into new submodule `Oscar.Serialization`, replace `@import_all_serialization_functions` by `using Oscar.Serialization` (#5024)

* Move serialization code into a submodule Oscar.Serialization

... and use that to get rid of @import_all_serialization_functions

* One line per export

* remove an import

* sort

* More re-imports

* no need to import load & save

* Update src/Serialization/LieTheory.jl

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

* Fix a test

* more

* oops

* Apply suggestions from code review

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

fingolfin committed:
Custom printing for FPGroupElem (#5035)

Abbreviate long words to fit into one screen line

fingolfin committed:
Rename IdealGens member 'gens' to 'gensBiPolyArray' (#5007)

... to make it easy to grep for. Preparation for merging BiPolyArray into IdealGens.

HechtiDerLachs committed:
Add support for spectral sequences in Cech cohomology (#4951)

* Widen _derived_pushforward to accept multi-gradings.

* WIP towards cached pushforwards.

* WIP on spectral sequences.

* Repair homogeneous Koszul complexes.

* Introduce induced maps on homogeneous Koszul complexes.

* Implement contravariant induced hom.

* Add tests.

* Adjust wip on derived pushforward.

* Implement induced maps in shifted complexes.

* Implement induced maps in tensor products.

* Implement induced maps in total complexes.

* Implement induced maps in views of complexes.

* Compute induced maps in truncated Cech complexes.

* Implement induced maps on strands.

* Use inclusion of strands.

* Add functionality for projection to (obvious) direct summands.

* Use the strand projections.

* Add some printing.

* Implement strand projections.

* Entries and maps on the first page.

* Produce entries on second page.

* Adapt to products of projective spaces.

* Improve caching.

* WIP on implementation of maps on 2nd page.

* Repair induced_ENC.

* Introduced lifted_kernel_generators for production of maps.

* Easen up lifting.

* Switch back to WeakKeyDicts.

* First working version of spectral sequences.

* Add some hacky tuning.

* Repair computations and remove debug messages.

* Fix bug in the shifted complexes.

* Fix bug in the induced_homs.

* Update tests.

* Add early return for production of maps.

* Use in-place operations.

* Clean up new Koszul complexes.

* Clean up the context object for pushforward.

* Clean up code for spectral sequences.

* Re-repair Koszul complexes.

* Add tests.

* Repair messed up routine for _derived_pushforward.

* Add a docstring for the spectral sequence.

* Actually include tests.

* Fix doctests.

* Remove deprecated code.

* Extend tests.

* Extend check_sanity.

* Clean up code.

* Implement covariant induced morphisms for hypercomplexes.

* Add first context object for the toric setting.

* Implement _minimal_exponent_vector naively.

* Allow use of ToricCtx in SpectralSequences.

* Add tests.

* Repair messed up hom-constructor.

* Drop support for sub for views.

* Revert faulty simplification.

* Fix doctests.

* Fix up iterators for ZZm gradings.

* Use better command to create matrix from rays.

* Regress code to 1.6 compatibility.

* Allow for fixed exponent vectors.

* Add some internal checks for SummandProjection.

* Clean up some code.

* Add method for setting exponent vector and algorithm keyword.

* Forward set_global_exponent_vector to spectral sequence.

* Adjust tests.

* Fixed uninitialized field.

* Code cleanup.

Co-authored-by: Max Horn <max@quendi.de>

* Fix tests.

---------

Co-authored-by: Max Horn <max@quendi.de>

ThomasBreuer committed:
add `permutation_character` (#5055)

HechtiDerLachs committed:
Migrate spectral sequences and derived pushforward to the hypercomplexes (#5054)

* Migrate spectral sequences and derived pushforward to the hypercomplexes.

* Move tests to the correct places, too.

benlorenz committed:
Drop support for julia versions older than current LTS (1.10) (#5051)

jamesnohilly committed:
Add more basic functionality for pc groups and their elements (#5026)

StevellM committed:
Move `QuadFormAndIsom` to source (#5046)

Promote the experimental package QuadFormAndIsom to src

fingolfin committed:
Tweak printing of symmetric & alternating groups (#5039)

fingolfin committed:
Remove refs_key (#5060)

Having a constant that is only used in two places seems like overkill.

fingolfin committed:
Tweak printing of AutomorphismGroup (#5036)

emikelsons committed:
Add is_noetherian, krull_dim, vector_space_dim methods (#4967)

fingolfin committed:
Support GAP 0.14 (#5049)

HereAround committed:
FTheoryTools documentation improvements (#5048)

* Update G4-flux documentation
* Remove su5_weierstrass_model_over_arbitrary_3d_base
* Remove su5_tate_model_over_arbitrary_3d_base
* Overhaul Weierstrass model documentation
* Add links via @ref

simonbrandhorst committed:
rational_points (#5068)

wdecker committed:
Add user interface for free resolutions over local rings (#5057)

lgoettgens committed:
Remove obsolete VERSION checks (#5058)

simonbrandhorst committed:
Irreducible components for schemes (#5065)

* irreducible components first version

benlorenz committed:
remove unnecessary runtests files from experimental packages (#5072)

these only have include statements and are bad for running the tests in parallel
also allows better tracking of per-file timings in the CI

thofma committed:
feat: support kernel of maps with Laurent polynomials as codomain (#5078)

lgoettgens committed:
Adapt Lie algebras to `vector_space_dim` (#4971)

fingolfin committed:
Replace dim by rank and krull_dim, cleanup (#5081)

HereAround committed:
FTheory: More documentation and related improvements (#5071)

* FTheoryTools: Improve documentation

* Remove method that turns Weierstrass model into Weierstrass model by returning the argument

* Code improvement

* Improve documentation text yet further

* Code optimization

* Overhaul hypersurface model documentation

* Code styling

* Support another hypersurface model constructor

* Improve hypersurface model documentation

* Provide more constructors for hypersurface models

* Line breaks - I believe there is convention for this

* Remove unnessary comments/update them

* Minor overhaul of the introduction page

* Do not use literature_model in documentation before it has been introduced

* Overhaul documentation examples

* Remove "# Examples" for consistency

* Stop support for sample_toric_variety

* Fix bugs

* Code improvements in the documentation

* Overhaul remainder of hypersurface models page

* Uniform styling of expression FTheoryTools

* Make links to Weierstrass, global Tate and hypersurface models across documentation

* Include literature models into the "basic" model documentation better

* Broad stroke refactoring of all remaining documentation

* Big strokes overhaul of Literature Model page

* Add introduction to literature models

* Begin adding a lot more detail to the literature model page

* Overhaul doc string of literature model constructor

* Further improvements for the literature model documentation

* More content for literature models documentation page

ThomasBreuer committed:
Change `root_of_unity(K::QQAbField, n::Int)` to return root corresponding to $\exp(2\pi i/n)$ (#5082)

fingolfin committed:
docs: fix type of some code blocks (#5047)

benlorenz committed:
booktest: try one more explicit cleanup for mongoc connection (#5084)

lkastner committed:
Use Laurent polynomial ring approach to speed up computation of `toric_ideal` (#5073)

---------
Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>
Co-authored-by: Max Horn <max@quendi.de>

fingolfin committed:
Clarify `all_blocks` docstring (#5088)

simonbrandhorst committed:
Require curves to be of pure dimension one. (#5066)

* Require curves to be of pure dimension one.

* dual projective curve

* carefully avoid computations of the radical ideal

---------

Co-authored-by: Matthias Zach <85350711+HechtiDerLachs@users.noreply.github.com>

paemurru committed:
singular_locus for projective schemes (#5067)

* singular_locus for projective schemes

* clean-up is_equidimensional
---------

Co-authored-by: anne <anne@krueger-berg.de>
Co-authored-by: Simon Brandhorst <brandhorst@math.uni-sb.de>

fingolfin committed:
Fix show for Oscar.Serialization.type_params (#5087)

afkafkafk13 committed:
fix issue #5090 (#5093)

* fix #5090

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

lgoettgens committed:
Move documentation to its own repository (#5086)

fingolfin committed:
Add `matrix_group(::GModule)` and `direct_sum(::Vector{<:ModuleFPHom})` (#5059)

lgoettgens committed:
Try to fix invariant theory booktests (again) (for nightly) (#5095)

lgoettgens committed:
Serialization: Add option to JSONSerializer to suppress saving refs (#5094)

fingolfin committed:
Tweak printing of (semi)direct & wreath product groups (#5040)

ThomasBreuer committed:
add a runtime check for the required Julia version (#5085)

joschmitt committed:
Add documentation for multi-sets and subset iterators to manual (#5091)

fingolfin committed:
Remove is_known(dim,...) methods (#5098)

Fe-r-oz committed:
📝 use `chain_complex` instead of `ComplexOfMorphisms` in documentation (#5050)

wdecker committed:
Adjust some surface code following PR #5066 (#5103)

HereAround committed:
Change output in doc test to remove random failures (cf. PR 5103) (#5107)

wdecker committed:
Addressing some (but not all) problems in issue #4860 (#5104)

emikelsons committed:
Rename some instances of `dim` (#5102)

ThomasBreuer committed:
add `induce` for `GSetBySubgroupTransversal` (#5092)

lgoettgens committed:
Revert "Set `OMP_NUM_THREADS=1` unconditionally (#4811)" (#5111)

This reverts commit 7972ad61db5d0306c9f5ae9ac6ee15b2bf0d39af.

fieker committed:
Fix bug in galois groups (#5109)

lgoettgens committed:
Speed up `degree(::GAPGroupClassFunction)` (#5113)

* Speed up `degree(::GAPGroupClassFunction)`

* Another similar change

* Fix `getindex(::GAPGroupClassFunction, AbstractVector{Int})`

lgoettgens committed:
Update AA to 0.46, Nemo to 0.51, Hecke to 0.37 (#5110)

* Update AA to 0.46, Nemo to 0.51, Hecke to 0.37

* Update explSL25.jlcon

jankoboehm committed:
fix krull_dim (#5115)

benlorenz committed:
disambiguate matrix construction for SubObjectIterator (#5116)

benlorenz committed:
Update Polymake.jl to v0.13, polymake to v4.14 (#5114)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>
Co-authored-by: Lars Kastner <lars.kastner@posteo.de>
Co-authored-by: Lars Kastner <lkastner@users.noreply.github.com>

lgoettgens committed:
Delete three obsolete files (#5128)

ThomasBreuer committed:
Fix multiplication of group cosets with group elements (#5123)

In the result, the parent of the representative must be equal to
the parent of the representative of the given coset,
otherwise the equality test for cosets does not work as expected.

lgoettgens committed:
Fix warning while doc building (#5135)

lgoettgens committed:
Fix race conditions in documentation deployment (#5134)

SoongNoonien committed:
Add `isinteger`, `is_rational` and `is_integral` for `QQAbFieldElem` (#5126)

fingolfin committed:
Micro-optimization: avoid one alloc per sparse_row call (#5136)

fingolfin committed:
Make `RootSpaceElem`, `DualRootSpaceElem` and `WeightLatticeElem` immutable (#5121)

lgoettgens committed:
Move `version-1-3-0-files` artifact to its own repo (#5127)

fingolfin committed:
Add `smallest_moved_point` and `largest_moved_point` (#5140)

Co-authored-by: Thomas Breuer <sam@math.rwth-aachen.de>

simonbrandhorst committed:
Set attributes for varieties. (#5118)

fingolfin committed:
Fix booktest in Julia nightly (#5142)

HereAround committed:
FTheoryTools: Yet more documentation changes (#5083)

* Fix rendering issue with matrix

* Add full explanation of model sections and friends to literature model documentation page

* Remove has_ and set_ functions to streamline code

* *FTheoryTools* to `FTheoryTools`

* Organize attributes of literature models more efficiently

* Extend macro and bring back doctests

* Attempts for fixes

* Another attempt to fix

* Stuff

* Some coool stuff

* Fix up type nightmare

* Fixes

* Fix failures

* Attempt to fix broken links

* Another attempt to fix links

* Another attempt

* Tester

* Attempt to fix links once more

* Avoid eval calls

* Add function signature to macro

* Further improvements on the macro

* Fix broken links to Literature Models

simonbrandhorst committed:
Some tweaks for Borcherds' method (#5144)

* virtually abelian in any rank

Co-authored-by: Max Horn <max@quendi.de>

fingolfin committed:
Make QQAbFieldElem & QQAbFieldGen immutable (#5120)

... to hopefully enable optimizations

HereAround committed:
Follow-up to #5083: Minor Doc Fixes (#5145)

HereAround committed:
[FTheoryTools] JSON files: model_sections -> tunable_sections

HereAround committed:
[FTheoryTools] Make links to Weierstrass Models page safer

HereAround committed:
[FTheoryTools] Make links to Global Tate Model page safer

HereAround committed:
[FTheoryTools] Make links to Hypersurface Model page safer

HereAround committed:
[FTheoryTools] Make links to Resolving F-Theory Models page safer

HereAround committed:
[FTheoryTools] Make links to Functionality For All F-Theory Models page safer

HereAround committed:
[FTheoryTools] Make links to Advanced (G4-flux) Methods safer

HereAround committed:
[FTheoryTools] Make link to G4-Fluxes page safer

HereAround committed:
[FTheoryTools] Doc strings of model sections etc. link to dedicated subsection on literature model page

HereAround committed:
[FTheoryTools] Provide dedicated page for QSM models

HereAround committed:
[FTheoryTools] Fix wrong heading level

HereAround committed:
[FTheoryTools] Move generating sections etc from literature models to functionality for all F-theory models

HereAround committed:
[FTheoryTools] Extend function generating macro to accept custom doc link

HereAround committed:
[FTheoryTools] Fix broken link

HereAround committed:
[FTheoryTools] Overhaul and restructure attributes of QSM models and their documentation

fingolfin committed:
Make some element structs immutable (#5124)

HereAround committed:
[FTheoryTools] Move printout controls to FamilyOfSpaces, as they only apply in this setting

ooinaruhugh committed:
Fix for issue #4865 (Output of `groebner_walk` is not reduced) (#5130)

* Fix non-reduced output of standard walk (fixes #4865)

* Add test case for reduced GB

* Update experimental/GroebnerWalk/test/standard_walk.jl

* Update experimental/GroebnerWalk/test/standard_walk.jl

Co-authored-by: Max Horn <max@quendi.de>

* fixes doctests

---------

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>
Co-authored-by: ederc <ederc@mathematik.uni-kl.de>
Co-authored-by: Max Horn <max@quendi.de>

aaruni96 committed:
Update changelog script (#4719)

Co-authored-by: Max Horn <max@quendi.de>

fingolfin committed:
Avoid fraction_field(::RationalFunctionField) (#5153)

For compatibility with upcoming AA changes

HereAround committed:
[FTheoryTools] Fix typo

HereAround committed:
[FTheoryTools] Overhaul of the generalities documentation

StevellM committed:
Fix printing for automorphism groups of `FinGenAbGroup` and `TorQuadModule` (#5132)

* Improve printing of automorphism groups and their elements for FinAbGenGroup and TorQuadModule

* Update src/Groups/abelian_aut.jl
---------

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

HereAround committed:
[FTheoryTools] Rough overhaul of the resolution page

antonydellavecchia committed:
Add serialization for Matrix Groups (#5122)

* Matrix group serialization

* adds MatrixGroupElem + test

* fix function signatures

* Apply suggestions from code review

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

* remove redundant line

* store with desc attr

* tests passing

* Apply suggestions from code review

Co-authored-by: Thomas Breuer <sam@math.rwth-aachen.de>

* add more tests for non generator MatrixGroupElem

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>
Co-authored-by: Thomas Breuer <sam@math.rwth-aachen.de>

wdecker committed:
More fixes on free resolutions (#5152)

* More fixes on free resolutions

* update test

* correction

HereAround committed:
[FTheoryTools] Move kbar^3 docu and source code to generalities

HereAround committed:
[FTheoryTools] Overhaul wording on page about resolving singularities

HereAround committed:
[FTheoryTools] Overhaul FamilyOfSpaces documentation page

HereAround committed:
[FTheoryTools] Better name for generalities page

HereAround committed:
[FTheoryTools] Optimize doc tests

HereAround committed:
[FTheoryTools] Fix broken links

HereAround committed:
[FTheoryTools] Improve MonteCarlo function (#5156)

---------
Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

fingolfin committed:
Don't allow `grade` on graded polynomial ring (#5080)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>
Co-authored-by: Johannes Schmitt <johannes.schmitt@ruhr-uni-bochum.de>

HereAround committed:
[FTheoryTools] Collect tuning code

HereAround committed:
[FTheoryTools] Absorb standard constructors into model constructor files

HereAround committed:
[FTheoryTools] Restructure source code of AbstractFTheoryModels

HereAround committed:
[FTheoryTools] Collect blowup functionality

HereAround committed:
[FTheoryTools] Doc string for display_all_literature_models appears on website

HereAround committed:
[FTheoryTools] Remove some comments

HereAround committed:
[FTheoryTools] Make calabi_yau_hypersurface attribute of AbstractFTheoryModels

lgoettgens committed:
Add another hash filter for nightly doctests (#5162)

HereAround committed:
[FTheoryTools] Overhaul adder functions (#5161)

HereAround committed:
[FTheoryTools] By default, verbosity printing is switched off

antonydellavecchia committed:
Update the labeled graph docs (#5169)

* add to the labeled graph docs

* adjust sentence

* add to doctest

* Apply suggestions from code review

Co-authored-by: Yue Ren <yue.ren.kl@gmail.com>

---------

Co-authored-by: Yue Ren <yue.ren.kl@gmail.com>

micjoswig committed:
slightly improved docs for graphs (#5171)

dependabot[bot] committed:
Bump actions/checkout from 4 to 5 (#5177)

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

emikelsons committed:
[FTheoryTools] Change some longer tests to be extra-long (#5178)

Fe-r-oz committed:
`homology` now works with `ComplexOfMorphisms{AbstractAlgebra.FPModule{FqFieldElem}}` (#5075)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

thofma committed:
Mention documentation preview in documentation of developer documentation of documentation (#5187)

joschmitt committed:
Adjust metioning of documentation preview in documentation (#5189)

* Revert "Mention documentation preview in documentation of developer documentation of documentation (#5187)"

This reverts commit 5f6b1e6828fbafc13d79d6e92c1ce713cd3a7c96.

* Remove information about preview link in CI

HechtiDerLachs committed:
Fix radical over integers (#5185)

* Fix up use of elimpart.

* Add tests.

* Fix doctests round 1.

antonydellavecchia committed:
small change to readme (#5190)

lgoettgens committed:
Parameterize change group type docstring (#5184)

lgoettgens committed:
Another round of (book) test changes for nightly (#5192)

ThomasBreuer committed:
fix `show` for `AutomorphismGroup` (#5165)

ThomasBreuer committed:
Use `orbit_via_Julia` generically (#5172)

HereAround committed:
Serialize FinGenAbGroupHom

ThomasBreuer committed:
Improve the documentation of G-sets (#5125)

- document `acting_group`, `action_function`, `action_homomorphism`,
  `orbit` for general G-sets,
- add methods for a `GSet` argument showing an error message that no method
  is available for the given type of G-sets,
- do not support `==` for arbitrary G-sets; for that, add a method that
  shows an error message and adjust some tests that assumed `==` for G-sets

HereAround committed:
[FTheoryTools] Drop support for cox_ring of FamilyOfSpaces

micjoswig committed:
phylogenetic trees: doc update (#5204)

HereAround committed:
[FTheoryTools] Improve and extend CI tests

HereAround committed:
[ToricVarieties] Serialize various maps
[FTheoryTools] Update QSMDB

LukasKuehne committed:
Fix a bug in the computation of matroid realization spaces with saturated defining ideal (#5182)

antonydellavecchia committed:
Add mixed graphs (graphs with both directed and undirected edges) (#5179)

* tidying for abstract graph

* update graph functions

* mnore to test

* update docs for mixed graphs

* fix doctest

* add docs + export components of mixed graph

* Apply suggestions from code review

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

* make copies of underlying components

* add docs for new edge iterators

* add missing examples

* missing exports

* better vertex handling

* fixes to not use copy

* typo in docs

* Update src/Combinatorics/Graphs/functions.jl

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

* add vertices through iterated add_vertex for mixed graph

* typo in docs for mixed_graph

* add to docs mixed_graph_from_edges

* Apply suggestions from code review

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

* minimizing exports

* fix regression in docs

* fix for doctests

* remove exports

* adjust vertices + remove mixed_graph_from edges

* more doctests fixes

* hopefully last one

* Update src/Combinatorics/Graphs/functions.jl

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>
Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

fingolfin committed:
Remove empty `src/Rings/integer.jl` (#5203)

thofma committed:
doc: adjust to changes in Nemo (#5194)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

HechtiDerLachs committed:
Add some developer documentation for is_known. (#5199)

* Add some developer documentation for is_known.

* Apply suggestions from code review

Co-authored-by: Tommy Hofmann <thofma@gmail.com>

* Wrap lines so that diffs for minor changes are easier to see

See also the following commits...

* Replace invalid LaTeX style quotes

* Oscar -> OSCAR

* Style tweaks

* clarify user-facing

---------

Co-authored-by: Tommy Hofmann <thofma@gmail.com>
Co-authored-by: Max Horn <max@quendi.de>

HechtiDerLachs committed:
Improve  `is_smooth` for affine schemes. (#5186)

* Improve method of is_one for IdealSheafOfSingularLocus.

* Switch to correct and more efficient implementation of is_smooth.

* Add support for simplifications of subquotient rings localized at rational points.

antonydellavecchia committed:
Adds support for multidimensional Array serialization (#5191)

* Serialization for multidim Arrays

* adds tests for MultiDimArray serialization

* fix typo

* fix ambiguities + add matrix tests

* updates from review

* updated to use slices and stacks

* fix unbounded types

* Apply suggestions from code review

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

* Update src/Serialization/containers.jl

* switch to use dims=1 to remain consistent with Matrix serialization

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

fingolfin committed:
Fix spelling: this is OSCAR, not Oscar (#5209)

Co-authored-by: Johannes Schmitt <johannes.schmitt@ruhr-uni-bochum.de>
Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

joschmitt committed:
Link to website in README for "How to cite OSCAR"

wdecker committed:
Intersection theory: Minor supplements to docu (#5208)

antonydellavecchia committed:
Add `is_acylic` function for directed graphs (#5211)

* adds is_acylic function for directed graphs

* Apply suggestions from code review

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

---------

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

HereAround committed:
[ToricVarieties] Introduce class_group_with_map and picard_group_with_map (#5064)

antonydellavecchia committed:
some bypass surgery for pmap (#5223)

* some by pass surgery

* Update experimental/Parallel/src/Parallel.jl

antonydellavecchia committed:
fixes regression in phylo serialization + tests (#5206)

antonydellavecchia committed:
Move phylogenetics (#5207)

* move-phylo

* add missing file

* delete old file

* phylogenetic tree as subtype of absrtact graph

* simplify changes

* drop whitespace changes

* fix printing

---------

Co-authored-by: Benjamin Lorenz <benlorenz@users.noreply.github.com>

benlorenz committed:
PolyhedralGeometry.visualize: add filename and backend keyword args (#5220)

benlorenz committed:
PolyhedralGeometry: (n_)cones now properly consider all cones (#5217)

HereAround committed:
[FTheoryTools] Reduce code duplication

joschmitt committed:
Use OSCAR<->GAP isomorphism of group in `stabilizer` (#5150)

fingolfin committed:
Fix `build_doc` issues with Julia 1.12, deduplicate printing docs, fix a broken link (#5232)

* Meta.parse() complained about an extra token after end of expression
* Enhance Oscar.build_doc() to help locate Meta.parse errors
* De-duplicate printing docs
* Also fix a broken reference to the 'Unicode printing' section
  (because we have two of those, it linked to ... somewhere in the
  docs for intersection theory ?!?

ThomasBreuer committed:
fix creation of `class_function` objects (#5227)

Some GAP functions may return plain lists where class functions
are expected.
Eventually this will get fixed in GAP but here is an Oscar fix
for the moment.

lgoettgens committed:
Bump `serialization-upgrade-tests` (#5238)

HechtiDerLachs committed:
Add `change_base_ring` for graded multivariate rings (#5240)

* Add change_base_ring for graded rings.

* Add tests.

antonydellavecchia committed:
adds export to Serialization Module for params_all_equal (#5230)

thofma committed:
feat: add rational_solutions for larger field (#5241)

- add `rational_solutions` to the documentation

RaphaelPaegelow committed:
Improving documentation for wreath Macdonald polynomials (#5021)

antonydellavecchia committed:
only upgrade code (#5249)

benlorenz committed:
add support for serializing linear programs over non-rational fields (#5246)

Co-authored-by: Antony Della Vecchia <vecchia@math.tu-berlin.de>
Co-authored-by: antonydellavecchia <antonydellavecchia@gmail.com>

wdecker committed:
Intersection theory: Some specifications (#5248)

* Intersection theory: Some specifications

* clarification

* get rid of misprints

YueRen committed:
Add documentation for graph visualization using Graphviz (#5193)

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>
Co-authored-by: Max Horn <max@quendi.de>

benlorenz committed:
graphs: add induced_subgraph (#5244)

SirToby25 committed:
Add difference polynomial rings and differential polynomial rings (#5212)

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

fingolfin committed:
Add type parameter to Matroid (#4933)

... and use this to make a bunch of code type stable.

HereAround committed:
[FTheoryTools] More content for the documentation

YueRen committed:
Minor changes to Groebner basis docu (#5253)

* CommutativeAlgebra: added missing indendation in docu

indentation necessary for "Note" block to be shown correctly

* CommutativeAlgebra: move `with_ordering` to the bottom of docs

As per its own documentation, it is a function that only advanced users should
use, so it shouldn't be the second function listed on the page.

* CommutativeAlgebra: move Default Orderings section

moved to bottom of Monomial Orderings page as per Christian's and Wolfram's preference

* CommutativeAlgebra: moved `default_ordering` examples to docstring

suggested by Wolfram

* CommutativeAlgebra: fixed docu wording

pointed out by Wolfram

* CommutativeAlgebra: added notes to `default_ordering` to Groebner bases sections

* CommutativeAlgebra: fixed typo in docstring

* Update docs/src/CommutativeAlgebra/GroebnerBases/groebner_bases.md

Co-authored-by: Johannes Schmitt <jschmitt@posteo.eu>

* Update docs/src/CommutativeAlgebra/GroebnerBases/groebner_bases_integers.md

Co-authored-by: Johannes Schmitt <jschmitt@posteo.eu>

* CommutativeAlgebra: added more references to docu

* CommutativeAlgebra: added OSCAR to Default Orderings header

---------

Co-authored-by: Johannes Schmitt <jschmitt@posteo.eu>

fingolfin committed:
Cleanup `src/Rings/hilbert.jl` (#5197)

* Cleanup src/Rings/hilbert.jl

- remove unused function filter2
- make PP_exponent const (oops)
- remove a bunch of unnecessary semicolons
- replace `XYZ::Vector{PP} = []` by `XYZ = PP[]`

* Use 'COND && break/continue/return VAL' pattern

* Replace loops with faster (!) loop comprehensions

* Cleanup show(::PP)

* Disable unused or redundant code

* Simplify HSNum_abbott

* Optimize project_indets

* Invert condiditon again

---------

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

HereAround committed:
Refine optional checks in toric varieties: introduce kwarg `completeness_check` (#5234)

benlorenz committed:
changelog: check more subprocess exit codes, full checkout to get tags and history (#5261)

HereAround committed:
FTheoryTools: Improve kwargs (#5214)

HereAround committed:
[FTheoryTools] We do not need consistency_checks for this G4-flux constructor

lgoettgens committed:
Remove some unreachable code (#5264)

lgoettgens committed:
Update to GAP.jl 0.15 (#5226)

joschmitt committed:
Allow permutation matrices over non-commutative rings (#5260)

antonydellavecchia committed:
Serialization upgrade test files version bump (#5255)

* use new files for lps and phylo trees

* test new files

* fix lie algebra serialization test

* Update Serialization-upgrade-test.jl

* reword testsets

* adjust lie algebra testset sentence as well

apturner committed:
[FTheoryTools] Updates to documentation on resolutions

HereAround committed:
[FTheoryTools] Remove _blowup_global and _blowup_global_sequence

HereAround committed:
Stop exporting _is_nontrivial

emikelsons committed:
Move `InfiniteDimensionError` to AA (#5158)

lkastner committed:
Use symbols instead of strings for variables in toric geometry (#4928)

---------

Co-authored-by: Martin Bies <HereAround@users.noreply.github.com>
Co-authored-by: Max Horn <max@quendi.de>
Co-authored-by: Martin Bies <MBies87@googlemail.com>
Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

HereAround committed:
[ToricVarieties] Make generic_section and functions derived thereby less random by providing a seed as kwarg (#5278)

---------

Co-authored-by: Miķelis Emīls Miķelsons <mikelis.emils.mikelsons@gmail.com>

lgoettgens committed:
Update Singular.jl to 0.26 (#5281)

ThomasBreuer committed:
fix the argument check in `conjugate_group` (#5276)

YueRen committed:
TropicalGeometry: changed tropical_semiring() docu (#5195)

added warning on mixing tropical and normal numbers due to
https://github.com/oscar-system/Oscar.jl/issues/4014

ederc committed:
Elimination GB handling after f4 (#5277)

* fixes GB handling after f4 if elimination ordering is chosen

* fixes ring generation

* adjusts test to new behaviour for elimination computations

* Update src/Rings/groebner/f4.jl

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

---------

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

emikelsons committed:
Implement chamber counting algorithm for toric line bundles

janikapeters committed:
BasisLieHightestWeight: Extend functionality to Demazure modules (#4570)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

HereAround committed:
[ToricVarieties] Specify and correct requirements under which the current cohomCalg implementation operates correctly

HechtiDerLachs committed:
Add a note about `underlying_scheme` (#5285)

jamesnohilly committed:
Consolidate `id_hom` and `identity_map` (#5247)

* Rename usages of identity_map

* Revert changes for Schemes

* Revert other renamings

* Update experimental/LieAlgebras/src/LieAlgebraHom.jl

* Update experimental/LieAlgebras/src/LieAlgebraModuleHom.jl

---------

Co-authored-by: Lars Göttgens <lars.goettgens@gmail.com>

fingolfin committed:
Groups: speed up == for identical inputs (#5289)

thofma committed:
chore: update AA to 0.47, Nemo to 0.52, and Hecke to 0.38 (#5282)

Co-authored-by: Lars Göttgens <lars.goettgens@rwth-aachen.de>

lgoettgens committed:
Version 1.6.0-DEV (#5294)

lgoettgens committed:
Fix most booktests on julia nightly (again) (#5297)

github-actions[bot] committed:
Update changelog for 1.5.0 release (#5243)

JohnAAbbott committed:
Jaa/hilbert clean up (#5262)

fingolfin committed:
Cleanup some docstrings in src/AlgebraicGeometry/RationalPoint/ (#5292)

fieker committed:
Permutation gmodules, hom, end for them, abelian groups (#5101)

* permutatio gmodules, hom, end for them, abelian groups

* progress: restrict endos

* speed up induction, needs AA PR

* should already be in by seperate PR

* "start" on char 0 MeatAxe.

The toplevel stuff is in gmodule_new:
 - rep from char
 - all rep (irreducible over Q)

Strategy:
 - try to split (into homogenous) permutation modules
 - split induced reps
in either way tries to keep dim < 100 (limit)

For G = transitive_group(11, 6) = M_11 finds all but 2 reps

Missing:
 - tensor wedge products to try
 - \otimes (in-fix) for characters
 - tensor_power and exterior_power for G-modules and characters
 - use of `reynold` to get end over ZZ/QQ via Plesken (or time against
   Gap
 - endo for tensor products, induced modules (if possible)
 - a sane strategy
 - condensation...

* fix induced module for GrpAb

* move MeatAxe into own file

* fix _rref and add gens

* first condensation

* Progress: M_11 is now working unassisted, but
 - the spinning needs improving
 - the strategy is bad
 - there is special action/ matrices for special modules missing
   (e.g. for permutation modules, don't use matrices)
 ...

But this "works":
```
G = transitive_group(11, 6)
Oscar.MeatAxe.gmodule_new(G)
```

also missing a function/ interface to get small subgroups of G to condense
with/ too

use a sane approach to the extra generator needed to condense Q[G]

* play with permutatio matrices (as a type) and more pitfalls with condensing

* better spinning

* it works...

* all groups up to order 80!

* various: (rep abelian groups, perm matrices)

 - _abs_irred_abelian gained an option to return the reps in
   cyclotomic fields rathern than the abelian closure
   Conversion from ab. closure -> cyclo -> number field is expensive
   (and unneccessary here)
 - PermMat: a matrix type to represent permutation matrices - without
   writing the matix down. Limited functionality, used for module homs.
   Optimized (?) for QQ
 - permutation_gmodule for the trivial group is a regular_gmodule.
   Regular gmodules have better endo/ center of endo
 - permutation_gmodules are now using the PermMat

* typo and use the better abelian reps

* view -> view_window

* idenpotent via pc-series

* propagte fac changes

* view_windows, part 2

janikapeters committed:
Fix `basis_lie_demazure` which wouldn't finish for some inputs (#5304)

HechtiDerLachs committed:
Another fix for minimal free resolutions (#5266)

* New implementation of minimize.

* Adapt docstring.

* Add tests.

* Fix completeness and length.

* Add new version which uses the already present resolution.

* Fix doctests.

* Fix doctests once more.

* Mimic the old behaviour of minimize.

* Restrict to the graded case only.

* Fix up printing.

* Fix doctests back.

* Sew the Frankenstein together with the old method.

* Catch edge case.

* Put the note back in.

HereAround committed:
Reduce yet more randomness in FTheoryTools (#5295)

lgoettgens committed:
Stabilize `is_principal_with_data` booktest (#5305)

aaruni96 committed:
Release notes: 2 level of changelog (#5271)

HereAround committed:
[FTheoryTools] Format code with julia formatter

ThomasBreuer committed:
request recursive GAP to Julia conversion where necessary (#5303)

Recursive conversion used to be the default for the direction
from GAP to Julia, but this may get changed in GAP.jl.
With the proposed change, the default becomes irrelevant.

HereAround committed:
Run julia formatter on toric varieties codes and enable the formatting tests

ThomasBreuer committed:
Add `order_bound` keyword argument to `subgroup_classes` (#5267)

HechtiDerLachs committed:
Resolve issue 5143.

wdecker committed:
Further corrections to free resolutions (#5307)

* Further corrections to free resolutions

* add test

simonbrandhorst committed:
Allow `discriminant_annihilator` as input for `enumerate_classes_of_lattices_with_isometry` (#5251)

* Allow discriminant_annihilator as input for enumerate_classes_of_lattices_with_isometry
---------
Co-authored-by: Stevell Muller <muller@cit-d0460cab9799.cit.uni-hannover.de>

HereAround committed:
Git blame ignores formatting toric varieties code (#5312)

HereAround committed:
Ignore change in git blame (#5313)

antonydellavecchia committed:
is lattice check clean up (#5229)

janikapeters committed:
Fix typo in documentation (#5324)

ederc committed:
AlgebraicSolving v0.10.0 (#5318)

HechtiDerLachs committed:
fix coordinates command in mpoly_localizations (Resolves issue #5316)

ThomasBreuer committed:
fix access to a `Fac` object (#5327)

* fix access to a `Fac` object

Apparently the `fac` field is (often) not defined.
According to the documentation,
it is recommended to use the object as an iterator.

* add tests for the discrete logarithm stuff

fingolfin committed:
Remove Oscar.aadir, .nemodir, .heckedir, .singulardir (#5335)

I could not see any reason why they might be beneficial. They
certainly make some code non-local, so harder to reason about
for both humans and tools like JET (not that it matters for JET
here, as they are used only for building the documentation).

fingolfin committed:
Merge branch 'master' into mh/origami-update
